docs/source/cpp/memory.rst

```diff
@@ -201,3 +201,261 @@ simply do::
    std::shared_ptr<arrow::Buffer> arbitrary_buffer = ... ;
    std::shared_ptr<arrow::Buffer> cpu_buffer = arrow::Buffer::ViewOrCopy(
       arbitrary_buffer, arrow::default_cpu_memory_manager());
+
+
+Memory Profiling
+================
+
+On Linux, detailed profiles of memory allocations can be generated using perf record,
+without any need to modify the binaries. These profiles can show the
+traceback in addition to allocation parameters (like size).
+
+.. TODO: This requires a debug build, right? Or maybe not if symbols in header file
```

Comment:
I _think_ you just need symbols - I was testing on the release build from conda-forge. But a debug build will let you set more fine-grained probes (e.g. on a specific line), though that's not really relevant here I guess.